packages/yourloops/components/date-pickers/trends-date-picker.tsx

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2022, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { FunctionComponent, MouseEventHandler } from 'react'
+
+import DateRangeIcon from '@mui/icons-material/DateRange'
+import InputAdornment from '@mui/material/InputAdornment'
+import TextField from '@mui/material/TextField'
+import IconButton from '@mui/material/IconButton'
+import NavigateBeforeIcon from '@mui/icons-material/NavigateBefore'
+import NavigateNextIcon from '@mui/icons-material/NavigateNext'
+import SkipNextIcon from '@mui/icons-material/SkipNext'
+import DialogRangeDatePicker from './dialog-range-date-picker'
+
+interface TrendsDatePickerProps {
+  atMostRecent: boolean
+  displayedDate: string
+  disabled: boolean
+  end: string
+  loading: boolean
+  maxDate: string
+  minDate: string
+  onBackButtonClick: MouseEventHandler<HTMLButtonElement>
+  onMostRecentButtonClick: MouseEventHandler<HTMLButtonElement>
+  onNextButtonClick: MouseEventHandler<HTMLButtonElement>
+  onResult: Function
+  start: string
+}
+
+export const TrendsDatePicker: FunctionComponent<TrendsDatePickerProps> = (props) => {
```

Comment:
It is not the same component. Trends date picker can select date range but daily cannot. I intentionally did not make it generic as I am convinced that it would juste make code harder to understand. We can discuss about it if you disagree :slightly_smiling_face: